File: php/tests/integration/test_app.php
#!/usr/bin/env php
<?php
/**
 * Integration test app for PHP SDK
 */

require_once __DIR__ . '/../../vendor/autoload.php';

use ErrorDebugger\ErrorDebugger;

if ($argc < 3) {
    fwrite(STDERR, "Usage: test_app.php <dsn> <token>\n");
    exit(1);
}

$dsn = $argv[1];
$token = $argv[2];

// Initialize SDK
ErrorDebugger::init([
    'dsn' => $dsn,
    'token' => $token,
    'service_name' => 'php-integration-test',
    'debug' => true
]);

// Trigger a test error
$exception = new Exception('Integration test error from PHP SDK');
ErrorDebugger::captureException($exception);

// Allow time for export
sleep(2);
ErrorDebugger::shutdown();
echo "PHP SDK test completed\n";

File: ruby/tests/integration/test_app.rb
#!/usr/bin/env ruby
# Integration test app for Ruby SDK

$LOAD_PATH.unshift File.expand_path('../../lib', __dir__)
require 'errordebugger'

if ARGV.length < 2
  warn 'Usage: test_app.rb <dsn> <token>'
  exit 1
end

dsn = ARGV[0]
token = ARGV[1]

# Initialize SDK
ErrorDebugger.init(
  dsn: dsn,
  token: token,
  service_name: 'ruby-integration-test',
  debug: true
)

# Trigger a test error
error = StandardError.new('Integration test error from Ruby SDK')
ErrorDebugger.capture_exception(error, test: 'integration', language: 'ruby')

# Allow time for export
sleep 2
ErrorDebugger.shutdown
puts 'Ruby SDK test completed'
